test/jestMatchers.ts

```diff
@@ -17,51 +17,80 @@ interface MatcherResult {
 
   if (received === expected) {
-    error = 'Expected image instances to be different';
+    errorString = 'Expected image instances to be different';
   } else if (received.width !== expectedImage.width) {
-    error = `Expected image width to be ${expectedImage.width}, but got ${received.width}`;
+    errorString = `Expected image width to be ${expectedImage.width}, but got ${received.width}`;
   } else if (received.height !== expectedImage.height) {
-    error = `Expected image height to be ${expectedImage.height}, but got ${received.height}`;
+    errorString = `Expected image height to be ${expectedImage.height}, but got ${received.height}`;
   } else if (received.depth !== expectedImage.depth) {
-    error = `Expected image depth to be ${expectedImage.depth}, but got ${received.depth}`;
+    errorString = `Expected image depth to be ${expectedImage.depth}, but got ${received.depth}`;
   } else if (received.colorModel !== expectedImage.colorModel) {
-    error = `Expected image color model to be ${expectedImage.colorModel}, but got ${received.colorModel}`;
+    errorString = `Expected image color model to be ${expectedImage.colorModel}, but got ${received.colorModel}`;
   } else {
-    rowsLoop: for (let row = 0; row < received.height; row++) {
-      for (let col = 0; col < received.width; col++) {
-        const receivedPixel = received.getPixel(col, row);
-        const expectedPixel = expectedImage.getPixel(col, row);
-        if (!this.equals(receivedPixel, expectedPixel)) {
-          error = `Expected pixel at (${col}, ${row}) to be [${expectedPixel.join(
-            ', ',
-          )}], but got [${receivedPixel.join(', ')}]`;
-          break rowsLoop;
+    if (error === 0) {
+      rowsLoop: for (let row = 0; row < received.height; row++) {
+        for (let col = 0; col < received.width; col++) {
+          const receivedPixel = received.getPixel(col, row);
+          const expectedPixel = expectedImage.getPixel(col, row);
+          if (!this.equals(receivedPixel, expectedPixel)) {
+            errorString = `Expected pixel at (${col}, ${row}) to be [${expectedPixel.join(
+              ', ',
+            )}], but got [${receivedPixel.join(', ')}]`;
+            break rowsLoop;
+          }
+        }
+      }
+    } else {
+      rowsLoop: for (let row = 0; row < received.height; row++) {
+        for (let col = 0; col < received.width; col++) {
+          for (let channel = 0; channel < received.channels; channel++) {
+            const receivedValue = received.getValue(col, row, channel);
+            const expectedValue = expectedImage.getValue(col, row, channel);
+            if (Math.abs(receivedValue - expectedValue) > error) {
+              errorString = `Expected value at (${col}, ${row}) to be in range [${
+                expectedValue - error
+              },${expectedValue + error}], but got ${receivedValue}`;
+              break rowsLoop;
+            }
+          }
         }
```

Comment:
Do we need two different loops? I would have only done one with a default error tolerance of zero

Reply:
I think I left it like that so that the error message displays the whole pixel when the error is zero, but I can change it.